src/main/scala/ru/d4team/chat/db/DBMigrator.scala

```diff
@@ -0,0 +1,51 @@
+package ru.d4team.chat.db
+
+import org.flywaydb.core.api.MigrationState
+import org.flywaydb.core.api.configuration.FluentConfiguration
+import ru.d4team.chat.config.PostgresConfig
+import zio._
+
+
+object DBMigrator {
+
+  def migrate: ZIO[PostgresConfig, Throwable, Unit] =
+    for {
+      config <- ZIO.service[PostgresConfig]
+      _ <- ZIO.logInfo(s"Starting the migration for host: ${config.url}")
+      count <- migrationEffect(config)
+      _ <- ZIO.logInfo(s"Successful migrations: $count")
+    } yield ()
+
+
+  private def migrationEffect(config: PostgresConfig): ZIO[Any, Throwable, Int] =
+    for {
+
+      flywayConfig <- ZIO.succeed(PostgresConfig.flywayConfig(config))
+      _ <- logValidationErrorsIfAny(flywayConfig)
+      _ <- ZIO.logInfo("Migrations validation successful")
+      count <- ZIO.succeed(flywayConfig.load().migrate().migrationsExecuted)
+
+      // fail for any statuses except success (in case of missing migration files, etc)
```

Comment:
Don't use comments when it can be displayed in method name. Put it in a separate method instead. Like this:

```
  private def checkStatesForSuccess(config: FluentConfiguration): IO[Error, List[Unit]] = {
    val migrationStates = config.load().info().all().toList.map(_.getState)
    ZIO.foreach(migrationStates) {
      case MigrationState.SUCCESS => ZIO.unit
      case e => ZIO.fail(new Error(s"Migration ${i.getDescription} status is not \"SUCCESS\": ${e.toString}"))
    }
  }
```

Reply:
you are absolutely right, comments have been removed